### PR TITLE
RavenDB-15311 Add validation for policy aggregation time

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/models/database/documents/timeValueEntry.ts [hunk 1]
```diff
@@ -85,6 +100,14 @@ class timeValueEntry {
         return this.amount() > 0;
     }
     
+    // isAboveMinimum() {
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/documents/timeValueEntry.ts]: ?

File: src/Raven.Studio/typescript/models/database/documents/timeValueEntry.ts [hunk 2]
```diff
@@ -60,6 +60,21 @@ class timeValueEntry {
         throw new Error("Unable to compare timeValueEntries: " + this.format() + " with " + second.format());
     }
     
+    isMultipleOf(previousPolicy:timeValueEntry) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/documents/timeValueEntry.ts]: space after :

File: src/Raven.Studio/typescript/models/database/documents/timeSeriesPolicy.ts [hunk 3]
```diff
@@ -125,16 +139,39 @@ class timeSeriesPolicy {
...
+                    message: "Minimum value for 'seconds' or 'months' time units is 1",
+                },
                 {
                     validator: (time: timeValueEntry) => !this.hasRetention() || time.isPositive(),
                     message: "Retention time must be greater than zero"
-                }
+                },
+                {
+                    validator: (time: timeValueEntry) => !this.hasRetention ||
+                        (this.retention().unit() !== 'second' && this.retention().unit() !== 'month') ||
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/documents/timeSeriesPolicy.ts:171]: can we simplify this condition?

File: src/Raven.Studio/typescript/models/database/documents/timeSeriesPolicy.ts [hunk 4]
```diff
@@ -125,16 +139,39 @@ class timeSeriesPolicy {
...
+                        time.amount() % 1 === 0,
+                    message: "Please use a whole number for 'seconds' or 'months' time units",
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/documents/timeSeriesPolicy.ts:173]: I wonder what we should write: whole number of integer. ;-)

File: src/Raven.Studio/typescript/models/database/documents/timeValueEntry.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/documents/timeValueEntry.ts]: done

File: src/Raven.Studio/typescript/models/database/documents/timeValueEntry.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/documents/timeValueEntry.ts]: removed  :)

File: src/Raven.Studio/typescript/models/database/documents/timeSeriesPolicy.ts (hunk 4 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/documents/timeSeriesPolicy.ts:173]: I think 'whole number' 
At least according to https://www.mathsisfun.com/whole-numbers.html

**Review by ml054 — APPROVED** [review]

**Pull request merged**